src/monmove.cpp

```diff
@@ -38,6 +40,69 @@ bool monster::wander()
     return ( goal == pos() );
 }
 
+bool monster::is_dangerous_field( const field &fd ) const
+{
+    // No fields = not dangerous (obviously)
+    if( fd.fieldCount() == 0 ) {
+        return false;
+    }
+
+    // Else check each field to see if it's dangerous to us
+    for( auto &fld : fd ) {
+        if( is_dangerous_field( fld.second ) ) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool monster::is_dangerous_field( const field_entry &entry ) const
+{
+    const field_id fid = entry.getFieldType();
+    // Check immunity towards fields
+    switch( fid ) {
+        case fd_smoke:
+        case fd_tear_gas:
+        case fd_toxic_gas:
+        case fd_relax_gas:
+        case fd_nuke_gas:
+            if (has_flag( MF_NO_BREATHE )) {
+                return false;
+            }
+            break;
```

Comment:
Issue with this; some of these are only dangerous at certain levels. For example thin smoke isn't dangerous, only densities of 2/3 are. As such returning ```!has_flag(MF_NO_BREATHE)``` will improperly return ```true``` without actually checking if the field itself is dangerous. If you've got a logical way to arrange it that can handle that fact I'm all for it though.